packages/core/content-manager/server/controllers/relations.js

```diff
@@ -1,64 +1,122 @@
 'use strict';
 
-const { prop, pick } = require('lodash/fp');
+const { prop, isEmpty, defaultsDeep } = require('lodash/fp');
+const { hasDraftAndPublish } = require('@strapi/utils').contentTypes;
 const { PUBLISHED_AT_ATTRIBUTE } = require('@strapi/utils').contentTypes.constants;
+const { transformParamsToQuery } = require('@strapi/utils/lib/convert-query-params');
 
 const { getService } = require('../utils');
+const { validateFindAvailable } = require('./validation/relations');
+
+const addWhereClause = (params, whereClause) => {
+  params.where = params.where || {};
+  if (params.where.$and) {
+    params.where.$and.push(whereClause);
+  } else {
+    params.where.$and = [whereClause];
+  }
+};
 
 module.exports = {
-  async find(ctx) {
+  async findAvailable(ctx) {
+    const { userAbility } = ctx.state;
     const { model, targetField } = ctx.params;
-    const { _component, ...query } = ctx.request.query;
-    const { idsToOmit } = ctx.request.body;
 
-    if (!targetField) {
-      return ctx.badRequest();
+    await validateFindAvailable(ctx.request.query);
+
+    const { component, entityId, idsToOmit, _q, ...query } = ctx.request.query;
+
+    const sourceModelUid = component || model;
+
+    const sourceModel = strapi.getModel(sourceModelUid);
+    if (!sourceModel) {
+      return ctx.badRequest("The model doesn't exist");
     }
 
-    const modelDef = _component ? strapi.getModel(_component) : strapi.getModel(model);
+    // permission check
+    if (entityId) {
+      const entityManager = getService('entity-manager');
+      const permissionChecker = getService('permission-checker').create({
+        userAbility,
+        model,
+      });
+
+      if (permissionChecker.cannot.read()) {
+        return ctx.forbidden();
+      }
+
+      const entity = await entityManager.findOneWithCreatorRoles(entityId, model);
 
-    if (!modelDef) {
-      return ctx.notFound('model.notFound');
+      if (!entity) {
+        return ctx.notFound();
+      }
+
+      if (permissionChecker.cannot.read(entity)) {
+        return ctx.forbidden();
+      }
     }
 
-    const attribute = modelDef.attributes[targetField];
+    const attribute = sourceModel.attributes[targetField];
     if (!attribute || attribute.type !== 'relation') {
-      return ctx.badRequest('targetField.invalid');
+      return ctx.badRequest("This relational field doesn't exist");
     }
 
-    const target = strapi.getModel(attribute.target);
+    const targetedModel = strapi.getModel(attribute.target);
+
+    const modelConfig = component
```

Comment:
Note for later we should have a common findConfiguration somewhere that does this if based on modelType